<page>
    <p2p-banner/>
    <h1>Play</h1>
    <alert type="warning">
        <text>This method replaces the deprecated <code>join_group_call</code> and <code>change_stream</code> methods.</text>
    </alert>
    <text>This method allows streaming audio or video to a group call or a private call.</text>
    <h3>Example</h3>
    <syntax-highlight mark="11-16">
        from pytgcalls import Client
        from pytgcalls import idle
        from pytgcalls.types import MediaStream
        ...

        app = PyTgCalls(client)
        app.start()

        ...  # Call API methods

        app.play(
            -1001185324811,
            MediaStream(
                'test.mp4',
            )
        )

        idle()
    </syntax-highlight>
    <separator/>
    <h2>Details</h2>
    <category-title noref="true">
        <ref src="method">PyTgCalls.<sb>play</sb></ref>()
    </category-title>
    <subtext>
        <text>Join in a group call or accept/request a private call to stream audio or video.</text>
        <br/>
        <category>
            <pg-title>PARAMETERS</pg-title>
            <subtext>
                <config id="ARG_CHAT_ID"/>
                <config id="ARG_INPUT_STREAM"/>
                <category-title><ref>config</ref>: Optional[Union[<docs-ref link="/PyTgCalls/Advanced Types/CallConfig">pytgcalls.types.CallConfig</docs-ref>, <docs-ref link="/PyTgCalls/Advanced Types/GroupCallConfig">pytgcalls.types.GroupCallConfig</docs-ref>]]</category-title>
                <subtext><text>Additional configuration for the call.</text></subtext>
            </subtext>
        </category>
        <br/>
        <category>
            <pg-title>RAISES</pg-title>
            <subtext>
                <config id="BASE_MTPROTO"/>
                <config id="GROUP_CALL_EXCEPTIONS"/>
                <config id="STREAM_EXCEPTIONS"/>
                <category-title>exception <ref>pytgcalls.exceptions.<sb>CallDeclined</sb></ref></category-title>
                <subtext><text>The call was declined by the user.</text></subtext>
                <category-title>exception <ref>pytgcalls.exceptions.<sb>CallDiscarded</sb></ref></category-title>
                <subtext><text>The call was discarded by the user while connecting.</text></subtext>
                <config id="TELEGRAM_EXCEPTION"/>
                <category-title>exception <ref>pytgcalls.exceptions.<sb>AlreadyJoinedError</sb></ref></category-title>
                <subtext><text>You are attempting to join in a call you're already a part of.</text></subtext>
                <config id="RTMP_EXCEPTION"/>
            </subtext>
        </category>
    </subtext>
</page>